tests/secrets_manager_test.py

```diff
@@ -49,6 +49,31 @@ def test_get_db_credentials(self, mock_boto3, mock_requests, mock_logger):
         assert actual_secret['username'] == 'db_user'
         assert actual_secret['password'] == 'db_pass'
 
+    @mock.patch('hanadb_exporter.secrets_manager.LOGGER')
+    @mock.patch('hanadb_exporter.secrets_manager.requests')
+    @mock.patch('hanadb_exporter.secrets_manager.boto3.session')
+    def test_get_db_credentials_imdsv2(self, mock_boto3, mock_requests, mock_logger):
+        mock_ec2_response = mock.Mock()
+        mock_requests.get.return_value = mock_ec2_response
```

Comment:
We would need to use `side_effects`: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect
> Using [side_effect](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect) to return a sequence of values:

So this would translate to something like (just some pseudo-code):
```
mock_ec2_unauthorized = mock.Mock()
mock_ec2_unauthorized.status_code = 401
mock_ec2_response = mock.Mock()
mock_ec2_response.json.return_value = json.loads('{"region":"test_region_imdsv2"}')
mock_requests.get.side_effects = [mock_ec2_unauthorized, mock_ec2_response]
mock_ec2_put = mock.Mock()
mock_ec2_put.content = "my-test-token"
mock_ec2_put.put.return_value = mock_ec2_put

# put all the asserts for each of the mock
```